buildsystem/generator/control_util.go

```diff
@@ -0,0 +1,23 @@
+package generator
+
+import (
+	"github.com/pkg/errors"
+	"github.com/yargevad/filepathx"
+)
+
+func withGlobMatches(patterns []string, op func(file string) error) error {
```

Comment:
evergreen uses this git-ignore parsing module module, and I think we should standardize on that

Reply:
I looked at [the package](https://godoc.org/github.com/sabhiram/go-gitignore) and the package interface seems to only be good at answering whether or not to ignore a particular file and is not actually able to find files. You would still have to roll your own `**` functionality to search for files.